// @leadercodes/modelsnpm/index.js
const express=require('express')
const app=express()
let {mongo_dev_url,mongo_prod_url}=require('./config/urls')

function Common() {
    //empty array to hold mongoose Schemas
    this.models = {};
    
}

Common.prototype.init =  function (mongoose) {

     mongoose.connect(
        app.get('env') === 'development' ? mongo_dev_url : mongo_prod_url , {

        useNewUrlParser: true,
        useUnifiedTopology: true,
        useCreateIndex: true,
        useFindAndModify: false,
    });
    mongoose.connection.on("connected", () => {
        console.log("connected");
    });
    const activityLogs = require('./models/activityLog')
    const googleContact = require('./models/googleContact');
    const quote = require('./models/quote');
    const systemWave = require('./models/systemWave');
    const chat = require('./models/chat');
    const company = require('./models/company')
    const contact = require('./models/contact')
    const conversation = require('./models/conversation')
    const course = require('./models/course')
    const customer = require('./models/customer')
    const file = require('./models/file')
    const hour = require('./models/hour')
    const invoice = require('./models/invoice')
    const lesson = require('./models/lesson')
    const lmsReview = require('./models/lmsReview')
    const lmsTeacher = require('./models/lmsTeacher')
    const lmsUser = require('./models/lmsUser')
    const product = require('./models/product')
    const school = require('./models/school')
    const signature = require('./models/signature')
    const superAdminUser = require('./models/superAdminUser')
    const tag = require('./models/tag')
    const team = require('./models/team')
    const user = require('./models/user')
    const wave = require('./models/wave')
    const webhook = require('./models/webhook')//webhook schema - Adina Schustal
    const deal = require('./models/deal')
    const workspace = require('./models/workspace')
    const answer = require('./models/answer')
    const article = require('./models/article')
    const buisness = require('./models/buisness')
    const calendarSettings = require('./models/calendarSettings')
    const card = require('./models/card')
    const category = require('./models/category')
    const departments = require('./models/departments')
    const event = require('./models/event')
    const eventsPageSettings = require('./models/eventsPageSettings')
    const hangout = require('./models/hangout')
    const helpCategory = require('./models/helpCategory')
    const notification = require('./models/notification')
    const organization = require('./models/organization')
    const project = require('./models/project')
    const question = require('./models/question')
    const task = require('./models/task')
    const teacher = require('./models/teacher')
    const test = require('./models/test')
    const ticket = require('./models/ticket')
    const copyContent = require('./models/copyContent')
    const draft = require('./models/draft')
    const notes = require('./models/notes')
    const source = require('./models/source')
    const tabHistory = require('./models/tabHistory')
    const funnel = require('./models/funnel')
    const logDetails = require('./models/logDetails')
    const priority = require('./models/priority')
    const sharedObject = require('./models/sharedObject')
    const staticLog = require('./models/staticLog')
    const taskStatus = require('./models/taskStatus')
    const userAnonymousDetails = require('./models/userAnonymousDetails')
    const add2contact = require('./models/add2contact')
    const mailTemplate=require('./models/mailTemplate')
    const notificationCalendar = require('./models/notificationCalendar')

    //add more model references here

    //This is just to make referencing the models easier within your apps, 
    // so you don't have to use strings. The model name you use here must match the name you used in the schema file
    this.models = {
        activityLogs: activityLogs,
        googleContact: googleContact,
        systemWave: systemWave,
        quote: quote,
        chat: chat,
        company: company,
        contact: contact,
        conversation: conversation,
        course: course,
        customer: customer,
        file: file,
        hour: hour,
        invoice: invoice,
        lesson: lesson,
        lmsReview: lmsReview,
        lmsTeacher: lmsTeacher,
        lmsUser: lmsUser,
        product: product,
        school: school,
        signature: signature,
        superAdminUser: superAdminUser,
        tag: tag,
        team: team,
        user: user,
        wave: wave,
        webhook: webhook, //Adina Schustal - webhook
        deal: deal,
        workspace: workspace,
        answer: answer,
        article: article,
        buisness: buisness,
        calendarSettings: calendarSettings,
        card: card,
        category: category,
        departments: departments,
        event: event,
        eventsPageSettings: eventsPageSettings,
        hangout: hangout,
        helpCategory: helpCategory,
        notification: notification,
        organization: organization,
        project: project,
        question: question,
        task: task,
        teacher: teacher,
        test: test,
        ticket: ticket,
        copyContent: copyContent,
        draft: draft,
        notes: notes,
        source: source,
        tabHistory: tabHistory,
        funnel: funnel,
        logDetails: logDetails,
        priority: priority,
        sharedObject: sharedObject,
        staticLog: staticLog,
        taskStatus: taskStatus,
        userAnonymousDetails: userAnonymousDetails,
        add2contact: add2contact,
        mailTemplate:mailTemplate,
        notificationCalendar:notificationCalendar

    }
}

var common = new Common();

module.exports = common;
