// @leadercodes/modelsnpm/models/buisness.js
const mongoose = require('mongoose')
const { express } = require('express');

//Tzofiya Cohen - for Buisness

const buisnessSchema = mongoose.Schema({
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User'
    },
    clientId: { type: String },
    dealerType: { type: String },
    numberDeals: { type: String },
    productionDate: { type: String },
    contacts: [
        { type: mongoose.Schema.Types.ObjectId, ref: 'Contact' }
    ],
    videos: [
        { type: mongoose.Schema.Types.ObjectId, ref: 'Video' }
    ],
    tags: [
        { type: mongoose.Schema.Types.ObjectId, ref: 'Tag' }
    ],
    invoices: [
        { type: mongoose.Schema.Types.ObjectId, ref: 'BuisnessInvoice' }
    ],
    products: [{
        type: mongoose.Schema.Types.ObjectId, ref: 'BuisnessProduct'
    }],
    landingPages: [
        { type: mongoose.Schema.Types.ObjectId, ref: 'LandingPage' }
    ],
    forms: [
        { type: mongoose.Schema.Types.ObjectId, ref: 'Form' }
    ],
    cards: [{
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Card'
    }],
    events: [{
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Event'
    }],
    blog: { type: mongoose.Schema.Types.ObjectId, ref: 'Blog' },
    chats: [{
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Chat'
    }],
    sites: [{
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Site'
    }],
    suspend: { type: Boolean, default: false },
    fullName: { type: String, default: '' },
    position: { type: String, default: '' },
    phone: {
        type: String,
        default: '',
        validate: {
            validator: function (v) {
                return /^$|^\d{10}$/.test(v);
            },
            message: props => `${props.value} is not a valid phone number!`
        },
    },
    name: { type: String, default: '' },
    socialmedias: {
        facebook: { type: String, default: '' },
        whatsapp: { type: String, default: '' },
        messenger: { type: String, default: '' },
        youtube: { type: String, default: '' },
        website: { type: String, default: '' }
    },
    address: { type: String, default: '' },
    numberAddress: { type: Number, default: 0 },
    city: { type: String, default: '' },
    state: { type: String, default: '' },
    zipcode: { type: String, default: '' },
    vat: { type: String, default: '' },
    imgProfile: { type: String, default: '' },
    imgLogo: { type: String, default: '' },
    stringBase: { type: String, default: '' },
    country: { type: String, default: '' },
    workHours: { type: Number, default: 0 },
    email: { type: String, default: '', match: /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/ },
    apiKey: { type: String },
    contactFacebookForm: { type: mongoose.Schema.Types.ObjectId, ref: 'ContactFacebookForms' },
    googleContacts: [{ type: mongoose.Schema.Types.ObjectId, ref: 'ContactGoogle' }],
})
module.exports = mongoose.model("Business", buisnessSchema)
