// @leadercodes/modelsnpm/models/chat.js
const mongoose = require('mongoose')

const chatSchema = mongoose.Schema({
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        require: true
    },
    Openning_Massage: String,
    endMessage: String,
    chatProfileImag: String,
    chat_name: { type: String, unique: true },
    name:String,
    chatTitle: String,
    chatSubTitle: String,
    chatColor: String,
    hederColor:String,
    agentColor:String,
    rulColor:String,
    bg_second_heder:String,
    bgBubble:String,
    borderBubble:String,
    colorBubble:String,
    iconColor:String,
    directionIframe:String,
    allowRegistration: Boolean,
    direction: String,
    autoMassage: String,
    convincingStatment: String,
    viewers: [
        { date: { type: String }, amount: { type: Number } }
    ],
    //message the user is getting when someone is trying chat with him.
    messageForSelfEmail: {
        type: String,
        default: 'Someone is trying to contact you in the chat'
    },
    formMessage: String,
    fieldName: String,
    fieldPhone: String,
    bubbleIcon: String,
    submitText: String,
    formMessageText: String,
    urlLink: String,
    createDate: { type: Date, default: Date.now },
    numberOfInquiries: Number,
    lastDateOfInterest: Date


})


module.exports = mongoose.model('Chat', chatSchema)
