// @leadercodes/modelsnpm/models/contact.js

const mongoose = require('mongoose')

const contactSchema = mongoose.Schema({
    name: { type: String },
    email: {
        type: String, required: true,
        match: /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
    },
    secondEmail: {
        type: String,
        match: /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
    },
    firstName: { type: String },
    lastName: { type: String },
    nickName: { type: String },
    address: { type: String },
    lat: { type: String },
    lng: { type: String },
    profile: { type: String },
    phone: {
        type: String,
        //  required: true,
        match: /\d{10}/
    },
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User'
    },
    displayToUser: { type: Boolean, default: true },
    starred: Boolean,
    thumbnail: String,
    numOfUnReadedWaves: Number,
    status: String,
    sourceUrl: String,
    leadOwner: String,
    leadSource: String,
    attached: String,
    customerType: String,
    companySize: String,
    companyName: String,
    companyAddress: String,
    gender: String,
    createDateAndTime: { type: Date, default: Date.now },
    bestTimeToCall: String,
    birthday: { type: Date, default: Date.now },
    telephon: String,
    mobileNumber: String,
    state: String,
    zipcode: String,
    website: String,
    whatsapp: String,
    linkedIn: String,
    facebook: String,
    instagram: String,
    youTube: String,
    active: Boolean,
    googleContact: { type: mongoose.Schema.Types.ObjectId, ref: 'GoogleContact' },
    addYourMedia: { type: String },

    officePhone: { type: String },
    //bestTimeToCall: { type: String },
    companyDescribe: { type: String },
    subscribe: { type: String },
    active: { type: Boolean, default: true },
    conversations: [
        { type: mongoose.Schema.Types.ObjectId, ref: 'Conversation' }
    ],
    Ticket: [
        { type: mongoose.Schema.Types.ObjectId, ref: 'Ticket' }
    ],
    quotes:
        [
            { type: mongoose.Schema.Types.ObjectId, ref: 'quotes' }
        ],

    source: [s = mongoose.Schema({
        id: { type: mongoose.Schema.Types.ObjectId },
        type: String
    })],
    leaderStatus: { type: String, default: "New" },
    leaderStatusContact: { type: String },
    socialMediaOption: [{
        _id: false,
        icon: { type: String },
        url: { type: String }
    }],
    rating: { type: Number },
    notes: [{ type: String , _id:false}],
    pipeline: { type: String },
    tags: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Tag' }], //Rivka- BOX


    //Tzofiya - quizes array for Quiz team
    quizes: [{
        testID: { type: String },
        done: { type: Boolean },
        cheating: { type: Boolean },
        kind: { type: String },
        test_name: { type: String },
        color: { type: String },
        questions: [{
            question: { type: String, required: true },
            answers: [{
                answerNumber: { type: Number },
                answer: { type: String },
                answerPicture: { type: String },
            }],
            openQuestion: { type: Boolean },
            correctAnswer: { type: Number, required: true },
            image: { type: String }
        }],
        // duration: { type: Number, required: true },
        time_days: { type: Number, required: true },
        time_hours: { type: Number, required: true },
        time_minutes: { type: Number, required: true },
        lastOpened: { type: Date },
        description: { type: String },
        displayVideo: { type: Boolean },
        video: { type: String },
        cvFile: { type: String },
        openTest: { type: Boolean },

        answersArray: [{
            questionNumber: { type: Number },
            studentAnswer: { type: String },
            is_correct: { type: Boolean },
            gradeOpenQuest: { type: Number },
            answerPicture: { type: String },
        }],
        multyQuestion_grade: { type: Number },
        grade: { type: Number },
        studentUserName: { type: String },
        backgroundImage: { type: String },
        creator: { type: String }

    }],
})

module.exports = mongoose.model('Contact', contactSchema)
