// @leadercodes/modelsnpm/models/course.js
const mongoose = require('mongoose')

const courseSchema = mongoose.Schema({
    school_id: { type: mongoose.Schema.Types.ObjectId, ref: 'School' },
    archive: { type: Boolean, default: false},
    archive_date: { type: Date },
    name: { type: String },
    subtitle: { type: String },
    category: { type: String },
    image: { type: String },
    price: { type: String },
    prev_price: { type: String },
    prev_price_time: {type: String },
    date_created: { type: String },
    course_info: [
      {
        name: { type: String },
        icon: { type: String }
      }
    ],
    teacher: { type: mongoose.Schema.Types.ObjectId, ref: 'LmsTeacher'},
    preview: { type: String },
    stars: { type: Number },
    views: { type: Number },
    weeks: { type: Number },
    students: [{ username: String,
                stars: Number }],
    lessons: [
        { type: mongoose.Schema.Types.ObjectId, ref: 'Lesson' }
    ],
    reviews:[{ type: mongoose.Schema.Types.ObjectId, ref: 'LmsReview' }],
    top_educators: [
      {
        bg: { type: String },
        image: { type: String },
        header: { type: String },
        content: { type: String },
        color: { type: String },
        discount: { type: String }
      }
    ],
    share:{
        instegram:{ type: String },
        facebook:{ type: String },
        whatsapp:{ type: String },
        twitter:{ type: String },
        linkedin:{ type: String },
        youtube:{ type: String },
        reddit:{ type: String },
    },
    belive:{
      text:{ type: String },
      auther:{ type: String },
      image:{ type: String },
    },
    description: [
      {
        header: { type: String },
        text: { type: String }
      }
    ],
    more_courses:{
      header:{ type: String },
      algorithm:{ type: String },
      items:{ type: Number }
    },
    show:{
      stars:{ type: Boolean },
      views:{ type: Boolean },
      prev_price:{ type: Boolean },
      price:{ type: Boolean },
      prev_price_time:{ type: Boolean },
      students:{ type: Boolean },
      weeks:{ type: Boolean },
      lessons:{ type: Boolean },
      students_review: { type: Boolean },
      more:{
        more_courses:{ type: Boolean },
        belive:{ type: Boolean },
        top_educators:{ type: Boolean },
        footer:{ type: Boolean },
      },
      share:{
        instegram:{ type: Boolean },
        facebook:{ type: Boolean },
        whatsapp:{ type: Boolean },
        twitter:{ type: Boolean },
        linkedin:{ type: Boolean },
        youtube:{ type: Boolean },
        reddit:{ type: Boolean },
      },
      description: { type: Boolean },
      reviews: { type: Boolean },
      teacher: { type: Boolean }
    },
    colors:{
      header: { type: String },
      name:{ type: String },
      subtitle:{ type: String },
      buy_course:{ type: String },
      more_courses:{ type: String },
      belive:{ type: String },
      top_educators:{ type: String }
    }
})

    

module.exports = mongoose.model('Course', courseSchema);
