// @leadercodes/modelsnpm/models/eventsPageSettings.js
const mongoose = require('mongoose')
// app eventsGallery 
// name -gila lorents
//Deputative values 
const EventsPageSettingsSchema = mongoose.Schema({

   
    eventsPageColor: {
        type: String,
        default: "#ad60ff"
    },
    eventsButtonColor: {
        type: String,
        default: "#51e7fb"
    },
    amountEventsInRow: {
        type: Number,
        default: 3
    },

    watchPreviousEvents: {
        type: Boolean,
        default: true
    },
    PreviousEvents: {
        type: Array,
        default: []
    },
    eventsPageAlignment: {
        type: String,
        default: 'left'
    },
    eventsPageImageOrVideo: {
        type: String,
        default: ""
    },
    eventsPageLogo: {
        type: String,
        default: ""
    },
    eventsPageTitle: {
        type: String,
        default: "Title Events"
    },
    eventsPageDescription: {
        type: String,
        default: "your description"
    },
    displayHeader: {
        type: Boolean,
        default: true
    },
    facebook: {
        type: Boolean,
        default: false
    },
    facebookLink: {
        type: String,
        default: ""
    },
    twitter: {
        type: Boolean,
        default: false
    },
    twitterLink: {
        type: String,
        default: ""
    },
    instagram: {
        type: Boolean,
        default: false
    },
    instagramLink: {
        type: String,
        default: ""
    },
    youtube: {
        type: Boolean,
        default: false
    },
    youtubeLink: {
        type: String,
        default: ""
    },
    name: {
        type: Boolean,
        default: true
    },
    email: {
        type: Boolean,
        default: true
    },
    phone: {
        type: Boolean,
        default: false
    },
    address: {
        type: Boolean,
        default: false
    },
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User'
    }

})


module.exports = mongoose.model('EventsPageSettings', EventsPageSettingsSchema)
