// @leadercodes/modelsnpm/models/example.js
const mongoose = require('mongoose')
const eventSchema = mongoose.Schema({
    title:{
        type:String
    },
    start: { 
        type: Date, 
        require: true, 
    },
    end: {
        type: Date,
    },
    rrule: {
        type:Object,
    },
    id:{
        type:mongoose.Schema.Types.ObjectId,
    },
    place: {
        type: String,
    },

    notifications:[
        {
            type:mongoose.Schema.Types.ObjectId,
            ref:'Notification'
        }
    ],

    createdDate:{
        type: Date
    },

    participants:
    [
        {
            type:mongoose.Schema.Types.ObjectId,
            ref:'Contact'
        }
    ],

    maxParticipants:{
        type: Number,
    },

    category:{
        type:mongoose.Schema.Types.ObjectId,
        ref:'Category' 
    },
    classNames:{
        type: String,
    },
    description: {
        type: String,
    },
    registrants:[{
        type:mongoose.Schema.Types.ObjectId,
        ref:'Contact'
    }],
    responsibles: [
        {
            type: String,
        }
    ],
    goals:{
        type: String,
    },
    aims:{
        type: String,
    },
    contactId:{
        type: String
    },
    taskId:{
        type: String,
    },
    user:{
        type:mongoose.Schema.Types.ObjectId,
        ref:'User' 
    },
    image:{
        type: String,
    },
    registrationURL:{
        type: String
    },
    confirmParticipation:[{
        type:mongoose.Schema.Types.ObjectId,
        ref:'Contact'
    }],
})

module.exports = mongoose.model('Example', eventSchema)
