// @leadercodes/modelsnpm/models/hour.js
//Sari kook - clocks
const mongoose = require("mongoose")
const { type } = require("os")

const hourSchema = mongoose.Schema({
    uid: { type: mongoose.Schema.Types.ObjectId, ref: "user" },
    user: { type: mongoose.Schema.Types.ObjectId, ref: "user" },
    startWork: { type: Date, default: Date.now },
    endWork: { type: Date },
    totalHour: { type: Date, default: "2021-04-12T00:00:00.837Z" },
    dateWork: { type: Date, default: Date.now },
    workspace: { type: String },
    project: { type: String },
    task: { type: String },
    card: { type: String },
    workspaceId: { type: mongoose.Schema.Types.ObjectId, ref: "Workspace" },
    projectId: { type: mongoose.Schema.Types.ObjectId, ref: "Project" },
    cardId: { type: mongoose.Schema.Types.ObjectId, ref: "Card" },
    taskId: { type: mongoose.Schema.Types.ObjectId, ref: "Task" },
    workspaceColor: { type: String },
    projectColor: { type: String },
    paymant: { type: String },
    amount: { type: Number },
    coin: { type: String },
    breaks: [{
        startBreak: { type: Date },
        endBreak: { type: Date },
        totalBreak: { type: Date }
    }],
    newDeatails: [{
        ip: { type: String },
        userAgent: { type: String },
        time: { type: Date },

    }],
    taskId: { type: String },
    description: { type: String },
    LocationWork: { type: String },
    // workspace: { type: mongoose.Schema.Types.ObjectId, ref: "Workspace1" },
    // project: { type: mongoose.Schema.Types.ObjectId, ref: "Project1" },
    // task: { type: mongoose.Schema.Types.ObjectId, ref: "Task1" }
})

module.exports = mongoose.model('Hour', hourSchema)
