// @leadercodes/modelsnpm/models/invoice.js
const mongoose = require("mongoose");
const { express } = require('express');
var dateFormat = require("dateformat");
// const autoIncrement = require('mongoose-auto-increment');
// var  = require("dateformat");

//Tzofiya Cohen - for Buisness

const invoiceForBuisness = new mongoose.Schema({
    buisness: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Buisness'
    },
    type: {
        type: String
    },
    invoiceNumber: {
        type: Number,
    },
    template: {
        type: String,
    },
    paypalLink: {
        type: String,
    },
    paymentStatus: {
        type: Boolean,
        default: false,
        required: true,
    },
    saveStatus: {
        type: String,
        default: "draft",
        required: true,
    },
    dueDate: {
        type: Date
    },
    date: {
        type: Date,
        default: dateFormat("paddedShortDate")
    },
    contact:
    {
        type: String,
    },
    products:
        [{
            id: {
                type: mongoose.Schema.Types.ObjectId,
                ref: 'BuisnessProduct'
            },
            amount: { type: Number },
            sum_product: {
                type: Number
            }
        }],
    publicNotes: {
        type: String,
    },
    paymentMethod: {
        type: String
    },
    imgLogo: {
        type: String
    },
    imgSignature: {
        type: String
    },
    imageFrame: {
        type: String
    },
    contactOneTime:
    {
        name: { type: String },
        address: { type: String, default: '' },
        phone: {
            type: String,
            default: '',
            validate: {
                validator: function (v) {
                    return /^$|^\d{10}$/.test(v);
                },
                message: props => `${props.value} is not a valid phone number!`
            },
        },
        email: { type: String, match: /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/ },
        flag: { type: Boolean, default: false }
    },

});

module.exports = mongoose.model('InvoiceForBuisness', invoiceForBuisness)
