// @leadercodes/modelsnpm/models/lesson.js
const mongoose = require('mongoose')

const lessonSchema = mongoose.Schema({
    course_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Course' },
    archive: { type: Boolean, default: false},
    archive_date: { type: Date },
    name: { type: String },
    subtitle: { type: String },
    files: [{ 
        name: { type: String },
        url: { type: String },
        size: { type: Number }
    }],
    time: { type: String },
    show: {
        stars: { type: Boolean },
        views: { type: Boolean },
        students: { type: Boolean },
        weeks: { type: Boolean },
        lessons: { type: Boolean },
        more_courses: { type: Boolean },
        belive: { type: Boolean },
        top_educators: { type: Boolean },
        footer: { type: Boolean },
        description: { type: Boolean }
    },
    settings: {
        autoplay: { type: Number },
        controls: { type: Number },
        loop: { type: Number },
        muted: { type: Number}
      }
})

module.exports = mongoose.model('Lesson', lessonSchema);
