// @leadercodes/modelsnpm/models/lmsUser.js
const mongoose = require('mongoose');

const lmsUserSchema = mongoose.Schema({
    email: { type: String, require: true, unique: true, match: /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/ },
    uid: { type: String, require: true },
    username: { type: String },
    suspend: { type: Boolean },
    photoURL: { type: String },
    isPro: { type: Boolean },
    school: { type: mongoose.Schema.Types.ObjectId, ref: 'School' },
    schoolsEnrolled: [{ type: mongoose.Schema.Types.ObjectId }],
    coursesEnrolled: [{
        courseId: { type: mongoose.Schema.Types.ObjectId },
        schoolId: { type: mongoose.Schema.Types.ObjectId },
        finishedLessons: [{ type: mongoose.Schema.Types.ObjectId }]
    }],
    profession: { type: String },
    about: { type: String },
    colors: {
        aboutStudent: { type: String },
        recomandCourses: { type: String }
    },
    leaderUser: { type: mongoose.Schema.Types.ObjectId, ref: 'User' }
})

module.exports = mongoose.model('LmsUser', lmsUserSchema);
