// @leadercodes/modelsnpm/models/organization.js
const mongoose = require('mongoose')

const organizationSchema = mongoose.Schema({
    name: { type: String },
    description: { type: String },
    userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
    teams: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Team' }],
    members: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Contact' }],
    colorStart: { type: String },
    colorEnd: { type: String }
})

organizationSchema.pre('save', function (next) {

    console.log(this.colorStart);
    console.log("colorEnd", this.colorEnd);
    //check if has color because save can be also after one save
    if (this.colorStart == undefined && this.colorEnd == undefined) {
        this.colorStart = getRandomColor();
        this.colorEnd = getRandomColor();
    }
    else {
        console.log("has color to organization");
    }
    next();
})
function getRandomColor() {
    var letters = '0123456789ABCDEF';
    var color = '#';
    for (var i = 0; i < 6; i++) {
        color += letters[Math.floor(Math.random() * 16)];
    }
    return color;
}
module.exports = mongoose.model("Organization", organizationSchema)
