// @leadercodes/modelsnpm/models/quote.js
const mongoose = require('mongoose')


const quoteSchema = mongoose.Schema({
    contactId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Contact'
    },
    userId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User'
    },
    name: String,
    nameYOrN: { type: Boolean, default: true },
    lastUpdateQuote: String,
    updateQuoteYOrN: Boolean,
    numOfViews: Number,
    testimonialPAYOrN: Boolean,
    testimonialPRYOrN: Boolean,
    testimonialTYOrN: Boolean,
    testimonialIYOrN: Boolean,
    OP1YOrN: Boolean,
    OP2YOrN: Boolean,
    OP3YOrN: Boolean,
    OP4YOrN: Boolean,
    imageYOrN: Boolean,
    aboutUsTitleYOrN: Boolean,
    aboutUsContainYOrN: Boolean,
    titleYOrN: Boolean,
    contactDetailsYOrN: Boolean,
    logo: String,
    LogoYOrN: Boolean,
    logoCNYOrN: Boolean,
    logoCompanyName: String,
    logoBorderRadiusLogo: String,
    logoBorderRadiusLogo1: String,
    logoBorderRadiusLogo2: String,
    logoBorderRadiusLogo3: String,
    logoBorderRadiusLogo4: String,
    logoWidth: String,
    logoHeight: String,
    logoBackgroundOnlyPng: String,
    logoSelectRadiuseView: String,
    contactDetailsTo: String,
    contactDetailsEmail: String,
    contactDetailsFrom: String,
    contactDetailsProposalName: String,
    contactDetailsFont: String,
    contactDetailsTextWeight: String,
    contactDetailsTextsize: String,
    contactDetailsAlignment: String,
    contactDetailsWidthText: String,
    contactDetailsLineHeight: String,
    contactDetailsColorText: String,
    titleTextTitle: String,
    titleFont: String,
    titleTextWeight: String,
    titleTextSize: String,
    titleAlignment: String,
    titleWidthText: String,
    titleLineHeight: String,
    titleColorText: String,
    aboutUsTitle: String,
    aboutUsTitleStyle: String,
    aboutUsAboutUs: String,
    aboutUsTextTitle: String,
    aboutUsFont: String,
    aboutUsTextWeight: String,
    aboutUsTextSize: String,
    aboutUsAlignment: String,
    aboutUsWidthText: String,
    aboutUsLineHeight: String,
    aboutUsColorText: String,
    imageImage: String,
    imageBackgroundOnlyPng: String,
    imageRectanglesWidth: String,
    imageRectanglesHeight: String,
    imageBorderRadiusRectangles: String,
    imageRectanglesColor: String,
    imageYN: Boolean,
    companyDetailsTitleCall: String,
    companyDetailsCall: String,
    companyDetailsTitleMail: String,
    companyDetailsMail: String,
    companyDetailsMail2: String,
    cccc: String,
    companyDetailsTitleWebsite: String,
    companyDetailsWebsite: String,
    companyDetailsBackgroundColor: String,
    companyDetailsCallYN: Boolean,
    companyDetailsMailYN: Boolean,
    companyDetailsWebsiteYN: Boolean,
    createdDate: { type: Date, default: Date.now },
    quillText: String,
    quillStyle: String,
    businessSignature: String,
    trimmedDataURL: String,
    trimmedDataURLYesOrNo: Boolean,
    contactDetailsDateYN: Boolean,
    contactDetailsToYN: Boolean,
    contactDetailsEmailYN: Boolean,
    contactDetailsFromYN: Boolean,
    emailsToSend: [],
    emailsToSendIndex: Number,
    emailsToSendI: Number,
    emailsToSendIndexServer: Number,
    pdf: String,
    quillYOrN: Boolean,
    customerSignatureText: String,

})


module.exports = mongoose.model('Quote', quoteSchema)
