// @leadercodes/modelsnpm/models/school.js
const mongoose = require('mongoose');

const schoolSchema = mongoose.Schema({
    username: { type: String },
    name: { type: String },
    studentsEnrolled: [{type: String }],
    title: { type: String },
    subtitle: { type: String },
    logo: { type: String },
    image: { type: String },
    courses: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Course' }],
    categories: [
      {
        name: { type: String },
        icon: { type: String },
        backcolor: { type: String }
      }
    ],
    CTA: {
      title: { type: String },
      text: { type: String }
    },

    learning:
    {
      header: { type: String },
      image: { type: String },
      info: [
        {
          id: { type: String },
          header: { type: String },
          text: { type: String }
        }
      ],
    },
    testimoinals:
      [
        {
          name: { type: String },
          image: { type: String },
          description: { type: String }
        }
      ],
    partners: [
      { type: String }
    ],
    getChoice:{
      header:{ type: String },
      algorithm:{ type: String },
      items:{ type: Number }
    },
    worldSelection:{
      header:{ type: String },
      algorithm:{ type: String },
      items:{ type: Number }
    },
    show: {
      categories: { type: Boolean },
      CTA: { type: Boolean },
      learning: { type: Boolean },
      testimoinal: { type: Boolean },
      partners: { type: Boolean },
      footer: { type: Boolean },
      getChoice: { type: Boolean },
      worldSelection: { type: Boolean },
      searchbar: { type: Boolean }
    },
    colors: {
      header: { type: String },
      title: { type: String },
      subtitle: { type: String },
      categories: { type: String },
      CTA: { type: String },
      learning: { type: String },
      partners: { type: String },
      footer: { type: String },
      testimoinal: { type: String },
      getChoice: { type: String },
      worldSelection: { type: String },
      searchbar: { type: String }
    },
    footer: [
      {
        name: { type: String },
        array: [{ type: String }],
        links: [{ type: String }]
      }
    ],
    footerBottom: {
      text: { type: String },
      share: {
        instegram: { type: String },
        facebook: { type: String },
        whatsapp: { type: String },
        twitter: { type: String },
        linkedin: { type: String },
        youtube: { type: String },
        reddit: { type: String },
        google: { type: String }
      },
      show: {
        instegram: { type: Boolean },
        facebook: { type: Boolean },
        whatsapp: { type: Boolean },
        twitter: { type: Boolean },
        linkedin: { type: Boolean },
        youtube: { type: Boolean },
        reddit: { type: Boolean },
        google: { type: Boolean },
      },
      aboutStudent:{ type: String }
    },
    choiceFavourite:{
      header: {type: String },
      algorithm: {type: String }
    },
    font: { type: String}
})

module.exports = mongoose.model('School', schoolSchema);
