// @leadercodes/modelsnpm/models/team.js
const mongoose = require('mongoose')
mongoose.set('useFindAndModify', false)
const Organization = require('./organization')
const User = require('./user')
const TeamController = ('../controllers/team')
//renana hacohen change 20.6 for hub

const teamSchema = mongoose.Schema({
    name: { type: String },
    logo: { type: String },
    admin: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
    teamLeader: { type: mongoose.Schema.Types.ObjectId, ref: 'Contact' },
    organization: { type: mongoose.Schema.Types.ObjectId, ref: 'Organization' },
    sharedObjects: [{
        objectId: { type: String },
        schemaName: { type: String },
        applicationName: { type: String }
    }],
    members: [{
        user: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
        contactMember: { type: mongoose.Schema.Types.ObjectId, ref: 'Contact' },
        sharedObjects: [{ type: mongoose.Schema.Types.ObjectId, ref: 'SharedObject' }]//renana change SharedObjects to SharedObject
    }]
})

teamSchema.pre('save', function (next) {
    //check if has logo because save can be also after one save
    if (this.logo == undefined) {
        this.logo = 'https://files.leader.codes/uploads/HesIhYWiU6gP73Sd0tWh2YW08dY2/img/1608490687372__logo1.png'
    }
    next()
})


module.exports = mongoose.model("Team", teamSchema)
